sdk/include/opentelemetry/sdk/trace/span_data.h

```diff
@@ -152,7 +153,37 @@ class SpanData final : public Recordable
    * @returns the attributes associated with the resource configured for TracerProvider
    */
 
-  const opentelemetry::sdk::resource::Resource &GetResource() const noexcept { return *resource_; }
+  const opentelemetry::sdk::resource::Resource &GetResource() const noexcept
+  {
+    if (resource_ == nullptr)
+    {
+      // this shouldn't happen as TraceProvider provides default resources
```

Comment:
Add assert here?

Reply:
Yes, that's one of the options. I just felt it's better to avoid asserting application if we can handle it gracefully. That also let us create unit-tests for ostream exporter ( which uses span-data) without depending on TracerProvider to pass the resources.